.github/workflows/job_cpu_functional_tests.yml

```diff
@@ -0,0 +1,124 @@
+name: CPU functional tests
+
+on:
+  workflow_call:
+    inputs:
+      runner:
+        description: 'Machine on which the tests would run'
+        type: string
+        required: true
+      image:
+        description: 'Docker image in which the tests would run'
+        type: string
+        required: false
+        default: null
+
+jobs:
+  CPU_Functional_Tests:
+    name: CPU functional tests
+    timeout-minutes: 25
+    runs-on: ${{ inputs.runner }}
+    container:
+      image: ${{ inputs.image }}
+    defaults:
+      run:
+        shell: bash
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      INSTALL_DIR: ${{ github.workspace }}/install
+      INSTALL_TEST_DIR: ${{ github.workspace }}/install/tests
+      PARALLEL_TEST_SCRIPT: ${{ github.workspace }}/install/tests/functional_test_utils/layer_tests_summary/run_parallel.py
+      PARALLEL_TEST_CACHE: ${{ github.workspace }}/install/tests/test_cache.lst
+    steps:
+      - name: Download OpenVINO package
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_package
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.INSTALL_TEST_DIR }}
+
+      # Needed as ${{ github.workspace }} is not working correctly when using Docker
+      - name: Setup Variables
+        run: |
+          echo "INSTALL_DIR=$GITHUB_WORKSPACE/install" >> "$GITHUB_ENV"
+          echo "INSTALL_TEST_DIR=$GITHUB_WORKSPACE/install/tests" >> "$GITHUB_ENV"
+          echo "PARALLEL_TEST_SCRIPT=$GITHUB_WORKSPACE/install/tests/functional_test_utils/layer_tests_summary/run_parallel.py" >> "$GITHUB_ENV"
+          echo "PARALLEL_TEST_CACHE=$GITHUB_WORKSPACE/install/tests/test_cache.lst" >> "$GITHUB_ENV"
```

Comment:
before we separated this job to a different workflow, we have not required echo these variables. What was changed??

Reply:
TL;DR: This step allowed the unified paths for jobs running outside and inside the Docker containers.

We use the introduced reusable workflows for four scenarios:
1. mac - GHA runners w/o the Docker image
2. linux - self-hosted with the Docker image 
3. linux - GHA runners w/o the Docker image

The first and third options do not use Docker so they are okay with `${{ github.workspace }}` path being set and used. However, the `${{ github.workspace }}` is incorrectly set for jobs running inside Docker containers - that is why we have a rather nasty hack with a hardcoded `/__w/openvino/openvino/` prefix for paths.
One way to unify this was to introduce this stage with variables setting: `${{ github.workspace }}` and `$GITHUB_WORKSPACE` are to be equal - it is true for jobs running w/o Docker, and `$GITHUB_WORKSPACE` being `/__w/openvino/openvino/` for jobs running inside Docker. 
If we set these variables, we can use the paths for both jobs: with and w/o Docker containers.